pytorch_lightning/trainer/training_loop.py

```diff
@@ -373,11 +366,10 @@ def _process_training_step_output_1_0(self, training_step_output, split_batch):
         # handle scalar return
         elif isinstance(training_step_output, torch.Tensor):
             loss = training_step_output
-            result["extra"] = {}
 
         # map to results under the hood
         result.minimize = loss
-        result.hiddens = hiddens
+        self.trainer.hiddens = hiddens
```

Comment:
is there any other spot we can stick this state to attaching more properties to the trainer?

Reply:
I'm sure there is. This could get cleaned up. I'm just keeping the current logic in this PR

https://github.com/PyTorchLightning/pytorch-lightning/blob/0e45220263f4e2045dfe7f68e3e0eaac0b2033d5/pytorch_lightning/trainer/training_loop.py#L701-L702

I would wait until more legacy code is removed though